app/controllers/self_enrollment_controller.rb

```diff
@@ -39,6 +41,12 @@ def enroll_self
 
   private
 
+  def check_early_student_join
+    return unless @course.is_a?(ClassroomProgramCourse) && @course.start.to_date > Time.zone.today
+
+    WikiExpertNotificationAlert.new(course: @course)&.send_email
```

Comment:
I think we also want to avoid  sending an unlimited number of these alerts, if many students join at once.

One single alert per course is probably fine, so I suggest an additional guard  clause that checks for an existing alert, before creating a new one.

Reply:
Okay